pkg/util/oc/oc_struct.go

```diff
@@ -421,42 +421,43 @@ func (o OC) GetYaml(t test.TestHelper, ns, kind, name string, checks ...common.C
 	return val
 }
 
-func (o OC) GetJson(t test.TestHelper, ns, kind, name string, checks ...common.CheckFunc) string {
+func (o OC) GetJson(t test.TestHelper, ns, kind, name, jsonPath string) string {
 	t.T().Helper()
-	var val string
+	var jsonString string
 	o.withKubeconfig(t, func() {
 		t.T().Helper()
-		val = shell.Execute(t, fmt.Sprintf("oc %s get %s/%s -ojson", nsFlag(ns), kind, name), checks...)
+		jsonString = shell.Execute(t, fmt.Sprintf("oc %s get %s %s -o jsonpath='%s'", nsFlag(ns), kind, name, jsonPath))
 	})
-	return val
+
+	return jsonString
 }
 
 // GetProxy returns the Proxy object from the cluster
 func (o OC) GetProxy(t test.TestHelper) *Proxy {
-	type ProxyResource struct {
-		Status *Proxy `json:"status"`
-	}
-
-	proxyResource := ProxyResource{}
+	proxy := o.GetJson(t, "", "proxy", "cluster", "{.status}")
 
-	proxyYaml := o.GetJson(t, "", "proxy", "cluster")
-	err := json.Unmarshal([]byte(proxyYaml), &proxyResource)
+	proxyStruct := &Proxy{}
```

Comment:
Why did you rename this from `proxy` to `proxyStruct`? It just makes the code harder to read.

Reply:
Done